packages/material-ui-utils/src/useForkRef.ts

```diff
@@ -2,21 +2,22 @@ import * as React from 'react';
 import setRef from './setRef';
 
 export default function useForkRef<Instance>(
-  refA: React.Ref<Instance> | null | undefined,
-  refB: React.Ref<Instance> | null | undefined,
+  ...refs: Array<React.Ref<Instance> | null | undefined>
 ): React.Ref<Instance> | null {
   /**
    * This will create a new function if the ref props change and are defined.
    * This means react will call the old forkRef with `null` and the new forkRef
    * with the ref. Cleanup naturally emerges from this behavior.
    */
   return React.useMemo(() => {
-    if (refA == null && refB == null) {
+    if (refs.every((ref) => ref == null)) {
       return null;
     }
     return (refValue) => {
-      setRef(refA, refValue);
-      setRef(refB, refValue);
+      refs.forEach((ref) => {
+        setRef(ref, refValue);
+      });
     };
-  }, [refA, refB]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
That's why it shouldn't be variadic.

Reply:
This warning is caused by the inability of the tool to verify if I passed all the dependencies. Since I use all the parameters provided to useForkRef, I know all of them should be in the dependencies (https://github.com/facebook/react/issues/14920#issuecomment-472018256).

Alternatively, if this really is a no-go for you, would a 4-parameter version of this function be more acceptable (it should cover all cases)?

```ts
function useForkRef<Instance>(
  refA: React.Ref<Instance> | null | undefined,
  refB: React.Ref<Instance> | null | undefined,
  refC?: React.Ref<Instance> | null | undefined,
  refD?: React.Ref<Instance> | null | undefined,
)
```

Reply:
But spreading is all of the sudden valid e.g. `useForkRef(...refs)`.

Reply:
Why is this not desirable, though?

Reply:
Because it breaks the rules of hooks

Reply:
I reverted to the original version and added two optional parameters. Please take a look.

Reply:
I don't think you want to see my point. Abandoning this discussion.

Reply:
Would I ask for it if I didn't want to? For me, the point of code review is to discuss the solution and highlight its good and bad sides.

If you think the solution sucks, I'm ok with that, as long as you explain your reasoning. This way I can either counter your arguments or simply learn something about the existing codebase and patterns.